drivers/wifi/simplelink_support.c

```diff
@@ -0,0 +1,616 @@
+		SET_STATUS_BIT(nwp.status, STATUS_BIT_IP_ACQUIRED);
+
+		/* Ip Acquired Event Data */
+		event_data = &netapp_event->Data.IpAcquiredV4;
+		sl_conn.ip_addr = event_data->Ip;
+
+		/* Gateway IP address */
+		sl_conn.gateway_ip = event_data->Gateway;
+
+		SYS_LOG_INF("\n[NETAPP EVENT] IP set to: IPv4=%d.%d.%d.%d, "
+		       "Gateway=%d.%d.%d.%d",
+		       SL_IPV4_BYTE(sl_conn.ip_addr, 3),
+		       SL_IPV4_BYTE(sl_conn.ip_addr, 2),
+		       SL_IPV4_BYTE(sl_conn.ip_addr, 1),
+		       SL_IPV4_BYTE(sl_conn.ip_addr, 0),
+
+		       SL_IPV4_BYTE(sl_conn.gateway_ip, 3),
+		       SL_IPV4_BYTE(sl_conn.gateway_ip, 2),
+		       SL_IPV4_BYTE(sl_conn.gateway_ip, 1),
+		       SL_IPV4_BYTE(sl_conn.gateway_ip, 0));
+
+		nwp.cb(NET_EVENT_IPV4_ADDR_ADD, &sl_conn);
+		break;
+
+	case SL_DEVICE_EVENT_DROPPED_NETAPP_IPACQUIRED_V6:
+		SET_STATUS_BIT(nwp.status, STATUS_BIT_IPV6_ACQUIRED);
+
+		for (i = 0; i < 4; i++) {
+			sl_conn.ipv6_addr[i] =
+			  netapp_event->Data.IpAcquiredV6.Ip[i];
+		}
+
+		SYS_LOG_INF("[NETAPP EVENT] IP Acquired: IPv6=");
+
+		for (i = 0; i < 3; i++) {
+			SYS_LOG_INF("%04x:%04x:",
+				    ((sl_conn.ipv6_addr[i] >> 16) & 0xffff),
+				    sl_conn.ipv6_addr[i] & 0xffff);
+		}
+
+		SYS_LOG_INF("%04x:%04x",
+			    ((sl_conn.ipv6_addr[3] >> 16) & 0xffff),
+			    sl_conn.ipv6_addr[3] & 0xffff);
+
+		break;
+
+	case SL_DEVICE_EVENT_DROPPED_NETAPP_IP_LEASED:
+		SET_STATUS_BIT(nwp.status, STATUS_BIT_IP_LEASED);
+		SET_STATUS_BIT(nwp.status, STATUS_BIT_IP_ACQUIRED);
+
+		sl_conn.sta_ip = netapp_event->Data.IpLeased.IpAddress;
+		SYS_LOG_INF("\n[NETAPP EVENT] IP Leased to Client: "
+			    "IP=%d.%d.%d.%d",
+			    SL_IPV4_BYTE(sl_conn.sta_ip, 3),
+			    SL_IPV4_BYTE(sl_conn.sta_ip, 2),
+			    SL_IPV4_BYTE(sl_conn.sta_ip, 1),
+			    SL_IPV4_BYTE(sl_conn.sta_ip, 0));
+
+		break;
+
+	case SL_DEVICE_EVENT_DROPPED_NETAPP_IP_RELEASED:
+		SYS_LOG_INF("\n[NETAPP EVENT] IP is released.");
+		break;
+
+	default:
+		SYS_LOG_ERR("\n[NETAPP EVENT] Unexpected event [0x%lx]",
+			    netapp_event->Id);
+		break;
+	}
+}
+
+/**
+ *  @brief SimpleLinkGeneralEventHandler
+ *
+ *  This handler gets called whenever a general error is reported
+ *  by the NWP / Host driver. Since these errors are not fatal,
+ *  the application can handle them.
+ *
+ *  @note See the CC3120/CC3220 NWP programmer's guide (SWRU455)
+ *	  section 17.9.
+ */
+void SimpleLinkGeneralEventHandler(SlDeviceEvent_t *dev_event)
+{
+	if (!dev_event) {
+		return;
+	}
+
+	SYS_LOG_INF("\n[GENERAL EVENT] - ID=[%d] Sender=[%d]",
+		    dev_event->Data.Error.Code,
+		    dev_event->Data.Error.Source);
+}
+
+/**
+ *  @brief SimpleLinkFatalErrorEventHandler
+ *
+ *  This handler gets called whenever a driver error occurs requiring
+ *  restart of the device in order to recover.
+ */
+void SimpleLinkFatalErrorEventHandler(SlDeviceFatal_t *fatal_err_event)
+{
+
+	switch (fatal_err_event->Id) {
+	case SL_DEVICE_EVENT_FATAL_DEVICE_ABORT:
+		SYS_LOG_ERR("\n[ERROR] - FATAL ERROR: "
```

Comment:
Can you check all sys_log lines and fix the \n, there are just too many of them to comment everyone.

Reply:
Same reason, it actually looks better with '\n' prefixes given the long SYS_LOG message prefix (due to the long CamelCase names of the "SimpleLink_XYZ_EventHander"s).